### PR TITLE
Poll delay

### PR DESCRIPTION
**Pull request by khennessy2**
Please note: this is only adding params to enable existing superclass consumer features. My concern was testing that the parameters work, not that the underlying features work 100% perfectly .

### CODE REVIEW

File: src/main/java/org/cdc/gov/sdp/queue/DatabaseQueueConsumer.java [hunk 1]
```diff
@@ -74,6 +74,9 @@ public DatabaseQueueConsumer(Endpoint endpoint, DataSource ds, Processor process
 		this.onConsumeFailed = "UPDATE " + tableName + " SET status = 'failed', attempts=attempts+1 where id=? ";
 		this.onConsume = "UPDATE " + tableName + " SET status = 'sent' where id=?";
 		this.onConsumeBatchComplete = "SELECT * FROM " + tableName;
+
+		this.setDelay(delay * 1000);// Converting from milliseconds to seconds
```
Comment (bccole) [inline src/main/java/org/cdc/gov/sdp/queue/DatabaseQueueConsumer.java]: Is the comment on this backwards?

File: src/test/resources/DatabaseQueueComponentTest-context.xml [hunk 2]
```diff
@@ -39,9 +39,13 @@
       <to id="producer_mock_to" uri="mock:mock_endpoint"/>
     </route>
     <route id="consumer_test_route">
-      <from id="consumer_from" uri="sdpqueue:message_queue?dataSource=sdpqDataSource"/>
+      <from id="consumer_from" uri="sdpqueue:message_queue?dataSource=sdpqDataSource&amp;initialDelay=7"/>
       <to id="consumer_mock_to" uri="mock:mock_endpoint"/>
     </route>
+    <route id="delayed_consumer_test_route">
```
Comment (bccole) [inline src/test/resources/DatabaseQueueComponentTest-context.xml]: It looks like both of these routes have a delay -- the first initial, the second between polls.  Can you rename them to something clearer for people coming across them?

File: src/test/java/org/cdc/gov/sdp/queue/DatabaseQueueComponentTest.java [hunk 3]
```diff
@@ -70,7 +73,119 @@ public void testQueueConsumer() throws Exception {
...
 
 			mock_endpoint.expectedMessageCount(rows_affected);
-			MockEndpoint.assertIsSatisfied(camelContext);
+			mock_endpoint.setResultWaitTime(5 * 1000);
+			mock_endpoint.assertIsNotSatisfied();
+
+			List<Map<String, Object>> lst = jdbcTemplate.queryForList(check_sent);
+			assertEquals(1, lst.size());
+		} finally {
+			jdbcTemplate.update(clear_dummy_data);
```
Comment (bccole) [inline src/test/java/org/cdc/gov/sdp/queue/DatabaseQueueComponentTest.java:85]: Is there a race condition here?  Might we have a situation where the data is written after the clear?

File: src/main/java/org/cdc/gov/sdp/queue/DatabaseQueueConsumer.java (hunk 1 above)
Comment (khennessy2) [reply src/main/java/org/cdc/gov/sdp/queue/DatabaseQueueConsumer.java]: Yep!

File: src/test/java/org/cdc/gov/sdp/queue/DatabaseQueueComponentTest.java (hunk 3 above)
Comment (khennessy2) [reply src/test/java/org/cdc/gov/sdp/queue/DatabaseQueueComponentTest.java:85]: I believe the assertIsNotSatisfied functions as a "wait" for the main test thread, but if that is not the case, then yes, there's a potential race condition.

File: src/test/resources/DatabaseQueueComponentTest-context.xml (hunk 2 above)
Comment (khennessy2) [reply src/test/resources/DatabaseQueueComponentTest-context.xml]: I will add a third well-named test route, but to clarify: by default (in a scheduledbatchpollingconsumer) the initialDelay is 1 second and delay is a half second; initialDelay must be set to 0 in order to test the delay parameter in the second test route.

**Review by bccole — APPROVED** [review]

**Pull request merged**